task/manager.go

```diff
@@ -870,6 +876,11 @@ func (m *Manager) updateRunning() {
 		running := task
 		pod, found := running.Reflect(&m.cluster)
 		if found {
+			err = m.ensureCollector(task, pod)
```

Comment:
**hint**: we don't know when each container will start so we need to ensure the collectors have been started every time we find a pod for a (running) task being processed.   The ensure is idempotent.